### PR TITLE
feat: added support for composite caching and test refactoring to split scenarios

### PR DESCRIPTION
**Pull request by camilomerlin**
added support for composite caching and refactored integration tests to separate the different scenarios into subpackages

### CODE REVIEW

File: cachestore/composite/cache_store.go [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+func (c *compositeCacheStore) Set(ctx context.Context, key string, value interface{}, expiration time.Duration) {
+	for _, cache := range c.delegates {
+		cache.Set(ctx, key, value, expiration)
+	}
+}
+
+func (c *compositeCacheStore) Get(ctx context.Context, key string, out interface{}) (bool, error) {
+	var err error
+	for _, cache := range c.delegates {
+		found, err2 := cache.Get(ctx, key, out)
```
Comment (pcu4dros) [inline cachestore/composite/cache_store.go]: I think it's better to rename this variable to c_err

File: cachestore/composite/cache_store.go [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+			err = err2
+		}
+	}
+	return false, err
+}
+
+func (c *compositeCacheStore) Delete(ctx context.Context, key string) error {
+	var err error
+	for _, cache := range c.delegates {
+		if err2 := cache.Delete(ctx, key); err2 != nil {
```
Comment (pcu4dros) [inline cachestore/composite/cache_store.go]: same here, rename err2 variable to c_err

File: cachestore/composite/cache_store.go (hunk 1 above)
Comment (camilomerlin) [reply cachestore/composite/cache_store.go]: Done, although I used `cErr` instead as the IDE didn't like `c_err` in Go :)

File: integration_tests/book_gorm_redis/test_starter.go [hunk 3]
```diff
@@ -0,0 +1,95 @@
+package book_gorm_redis
+
+import (
+	"context"
+	redis2 "github.com/go-redis/redis"
```
Comment (daltemen) [inline integration_tests/book_gorm_redis/test_starter.go]: please avoid use default alias like `redis2`

File: integration_tests/book_gorm_redis/testdomain/system_instance.go [hunk 4]
```diff
@@ -0,0 +1,18 @@
+package testdomain
+
+import (
+	"context"
+	"github.com/jinzhu/gorm"
+	"github.com/merlinapp/datarepo-go"
+	"github.com/merlinapp/datarepo-go/cachestore/stats"
+	stats2 "github.com/merlinapp/datarepo-go/repo/stats"
```
Comment (daltemen) [inline integration_tests/book_gorm_redis/testdomain/system_instance.go]: the same here

File: integration_tests/bookcategory_gorm_numericid/test_starter.go [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+import (
+	"context"
+	_ "github.com/go-sql-driver/mysql"
+	"github.com/merlinapp/datarepo-go"
+	"github.com/merlinapp/datarepo-go/cachestore/memory"
+	"github.com/merlinapp/datarepo-go/cachestore/stats"
+	"github.com/merlinapp/datarepo-go/integration_tests"
+	"github.com/merlinapp/datarepo-go/integration_tests/bookcategory_gorm_numericid/testdomain"
+	"github.com/merlinapp/datarepo-go/integration_tests/model"
+	gorm2 "github.com/merlinapp/datarepo-go/repo/gorm"
```
Comment (daltemen) [inline integration_tests/bookcategory_gorm_numericid/test_starter.go]: the same here

File: integration_tests/booktype_gorm_composite/test_starter.go [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+	_ "github.com/go-sql-driver/mysql"
+	"github.com/merlinapp/datarepo-go"
+	"github.com/merlinapp/datarepo-go/cachestore/composite"
+	"github.com/merlinapp/datarepo-go/cachestore/memory"
+	"github.com/merlinapp/datarepo-go/cachestore/nocache"
+	"github.com/merlinapp/datarepo-go/cachestore/stats"
+	"github.com/merlinapp/datarepo-go/integration_tests"
+	"github.com/merlinapp/datarepo-go/integration_tests/booktype_gorm_composite/testdomain"
+	"github.com/merlinapp/datarepo-go/integration_tests/model"
+	gorm2 "github.com/merlinapp/datarepo-go/repo/gorm"
```
Comment (daltemen) [inline integration_tests/booktype_gorm_composite/test_starter.go]: same here

File: integration_tests/booktype_gorm_composite/test_starter.go [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+
+func startSystemForIntegrationTests() *testdomain.SystemInstance {
+	if testInstance != nil {
+		return testInstance
+	}
+
+	db := integration_tests.TestConnectionFactory()
+	db.LogMode(true)
+
+	memoryCacheStore := memory.NewFreeCacheInMemoryStore(1 * 1024 * 1024)
```
Comment (daltemen) [inline integration_tests/booktype_gorm_composite/test_starter.go]: maybe convert those integer values to variables to know what it is
```suggestion
	memoryCacheStore := memory.NewFreeCacheInMemoryStore(1 * 1024 * 1024)
```

File: integration_tests/booktype_gorm_memory/test_starter.go [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+import (
+	"context"
+	_ "github.com/go-sql-driver/mysql"
+	"github.com/merlinapp/datarepo-go"
+	"github.com/merlinapp/datarepo-go/cachestore/memory"
+	"github.com/merlinapp/datarepo-go/cachestore/stats"
+	"github.com/merlinapp/datarepo-go/integration_tests"
+	"github.com/merlinapp/datarepo-go/integration_tests/booktype_gorm_memory/testdomain"
+	"github.com/merlinapp/datarepo-go/integration_tests/model"
+	gorm2 "github.com/merlinapp/datarepo-go/repo/gorm"
```
Comment (daltemen) [inline integration_tests/booktype_gorm_memory/test_starter.go]: same here

File: integration_tests/book_gorm_redis/test_starter.go (hunk 3 above)
Comment (camilomerlin) [reply integration_tests/book_gorm_redis/test_starter.go]: some of these are needed as I already have a `redis` package coming from somewhere else - I'll update the name though :)

**Review by daltemen — APPROVED** [review]

**Review by pcu4dros — APPROVED** [review]

**Pull request merged**